SecondStage/module28/src/test/java/test/TestBase.java

```diff
@@ -0,0 +1,24 @@
+package test;
+
+import driver.DriverSingleton;
+import org.openqa.selenium.WebDriver;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import util.TestListener;
+
+@Listeners ({TestListener.class})
+public class TestBase {
+
+    protected WebDriver driver;
+
+    @BeforeMethod
```

Comment:
To re-initialize webDriver and close and open browser on each test method is very inefficient way. I'd suggest to have methods that will start from a home page and return it after passing. So you can use same session to speed up test execution